lib/utils/Utils.ts

```diff
@@ -0,0 +1,343 @@
+ */
+export function generateRandomString(length: number): string {
+  const result = [];
+  const alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_';
+  for (let i = 0; i < length; i++) {
+    result.push(alphabet.charAt(Math.floor(Math.random() * alphabet.length)));
+  }
+  return result.join('');
+}
+
+/**
+ * Converts time (h:m:s) to seconds.
+ *
+ * @param time
+ * @returns seconds
+ */
+export function timeToSeconds(time: string) {
+  const params = time.split(':').map((param) => parseInt(param));
+  switch (params.length) {
+    case 1:
+      return params[0];
+    case 2:
+      return params[0] * 60 + params[1];
+    case 3:
+      return params[0] * 3600 + params[1] * 60 + params[2];
+    default:
+      throw new Error('Invalid time string');
+  }
+}
+
+/**
+ * Converts strings in camelCase to snake_case.
+ *
+ * @param string The string in camelCase.
+ */
+export function camelToSnake(string: string) {
+  return string[0].toLowerCase() + string.slice(1, string.length).replace(/[A-Z]/g, (letter) => `_${letter.toLowerCase()}`);
+}
+
+/**
+ * Checks if a given client is valid.
+ *
+ * @param client
+ * @returns
+ */
+export function isValidClient(client: string) {
+  return VALID_CLIENTS.has(client);
+}
+
+/**
+ * Throws an error if given parameters are undefined.
+ * @param params
+ */
+export function throwIfMissing(params: object) {
+  for (const [ key, value ] of Object.entries(params)) {
+    if (!value)
+      throw new MissingParamError(`${key} is missing`);
+  }
+}
+
+/**
+ * Turns the ntoken transform data into a valid json array
+ * @param data
+ */
+export function refineNTokenData(data: string) {
+  return data
+    .replace(/function\(d,e\)/g, '"function(d,e)').replace(/function\(d\)/g, '"function(d)')
+    .replace(/function\(\)/g, '"function()').replace(/function\(d,e,f\)/g, '"function(d,e,f)')
+    .replace(/\[function\(d,e,f\)/g, '["function(d,e,f)').replace(/,b,/g, ',"b",')
+    .replace(/,b/g, ',"b"').replace(/b,/g, '"b",').replace(/b]/g, '"b"]')
+    .replace(/\[b/g, '["b"').replace(/}]/g, '"]').replace(/},/g, '}",')
+    .replace(/""/g, '').replace(/length]\)}"/g, 'length])}');
+}
+
+/**
+ *
+ */
+export function uuidv4() {
+  if (getRuntime() === 'node') {
+    return Reflect.get(module, 'require')('crypto').webcrypto.randomUUID();
+  }
+
+  if (globalThis.crypto?.randomUUID()) {
+    return globalThis.crypto.randomUUID();
+  }
+
+  // See https://stackoverflow.com/a/2117523
+  return '10000000-1000-4000-8000-100000000000'.replace(/[018]/g, (cc) => {
+    const c = parseInt(cc);
+    return (c ^ window.crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16);
+  });
+
+
+}
+
+export type Runtime = 'node' | 'deno' | 'browser';
+
+/**
+ *
+ */
+export function getRuntime(): Runtime {
+  if ((typeof process !== 'undefined') && (process?.versions?.node))
+    return 'node';
+  if (Reflect.has(globalThis, 'Deno'))
+    return 'deno';
+  return 'browser';
+}
```

Comment:
Maybe make this check compile time?

Reply:
I think most bundling tools nowadays support `package.exports` and as long as that works, well, it is just as simple as importing it.

Reply:
I still feel it might be beneficial to not bundle everything and let the user handle that. (For example, if someone wants to use just the parsers and not the rest of the modules in a project)

Reply:
Okay, fair point.